src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -45,10 +52,18 @@ public SubstringConstraint(string expected) : base(expected)
         /// <returns>True for success, false for failure</returns>
         protected override bool Matches(string actual)
         {
-            if (this.caseInsensitive)
-                return actual != null && actual.ToLower().IndexOf(expected.ToLower()) >= 0;
-            else
-                return actual != null && actual.IndexOf(expected) >= 0;
+            return actual != null && actual.IndexOf(expected, comparisonType) >= 0;
```

Comment:
To keep the same behavior as `IgnoreCase` and `Using` on `EqualityComparer`, we'll need the `comparisonType` to be nullable in case no one calls `Using`. Then in here, if `comparisonType` is not null, it overrides everything else and returns exactly what you have here. If it is null, we need to run the original `caseInsensitive` code. (/cc @CharliePoole)